### PR TITLE
Add tests, javadocs and multilingual support 

### PR DESCRIPTION
**Pull request by hongshui**
REMARKS:
- There are a lot of files changed, but most of changes are just new javadocs or renamed methods
- There were also some changes to the packaging
- Two languages are now supported: english and polish
- New tests were added to accomodate above changes, mostly in board package

### CODE REVIEW

File: estimates.adoc [hunk 1]
```diff
@@ -17,9 +17,21 @@
...
-|clean-up + code quality tools
-|+ logs
-|+ find winner (sequence-based implementation)
+|clean-up
+|+ more scenarios to scripted game
+|+ code quality tools
+|04.01
+|--
+|--
+|--
```
Comment (KrzysiekJodlowski) [inline estimates.adoc]: If there's nothing happening there's no need to mention it;)

File: src/main/java/com/github/hongshui/ox/board/BoardObserver.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	 */
+	void onWin();
+
+	/**
+	 * This method is invoked when there is draw on board.
+	 */
+	void onDraw();
+
+	/**
+	 * This method is invoked when there is neither winner nor draw.
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/board/BoardObserver.java]: Lot of "this" (just an observation), descriptions are not bad but could be better (just a little note).

File: src/main/java/com/github/hongshui/ox/board/Lines.java [hunk 3]
```diff
@@ -7,39 +7,29 @@
...
-    private final Map<Integer, Line> rows = new HashMap<>();
-    private final Map<Integer, Line> columns = new HashMap<>();
-    private final Map<Integer, Line> diagonal = new HashMap<>();
-    private final Map<Integer, Line> antiDiagonal = new HashMap<>();
-    private int insertedFields;
-    private boolean isThereAWinner = false;
+    final Map<Integer, Line> rows = new HashMap<>();
+    final Map<Integer, Line> columns = new HashMap<>();
+    final Map<Integer, Line> diagonal = new HashMap<>();
+    final Map<Integer, Line> antiDiagonal = new HashMap<>();
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/board/Lines.java]: So, why not private? I'm just curious:)

File: src/main/java/com/github/hongshui/ox/board/ObservableBoard.java [hunk 4]
```diff
@@ -1,7 +1,5 @@
 package com.github.hongshui.ox.board;
 
-import com.github.hongshui.ox.commons.BoardObserver;
-
 import java.util.LinkedList;
 import java.util.List;
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/board/ObservableBoard.java]: You're not impose anything in that interface, why is it an interface?

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java [hunk 5]
```diff
@@ -1,62 +1,54 @@
...
-	static RandomizedUI create(SetupUI ui, int borderLength, boolean doWait) {
-		return new RandomizedUI(ui, RandomFieldSelector.generate(borderLength), doWait);
+	static RandomDataSource create(UserInterface ui, int borderLength, boolean doWait) {
+		return new RandomDataSource(ui, RandomFieldSelector.generate(borderLength), doWait);
 	}
 
 	@Override
 	public Players acquirePlayers() {
-		Players players = new Players(this);
+		Players players = new Players(ui);
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: Hmm, passing ui to Players, after review of my own ideas the warning light turns on;) Is it necessary?

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java [hunk 6]
```diff
@@ -1,62 +1,54 @@
...
 	private int visiblySelect(int number) {
 		if (doWait) { visiblyWait(); }
-		numberSelected(number);
+		ui.print((number + 1) + "\n");
 		return number;
 	}
 
-	private void visiblyWait() {
+	void visiblyWait() {
 		try { TimeUnit.MILLISECONDS.sleep(400); }
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: Thats interesting;)

File: src/main/java/com/github/hongshui/ox/game/Game.java [hunk 7]
```diff
@@ -1,33 +1,53 @@
...
+	private void setupTheGame() {
+		setup.setupLanguage();
+		ui.print(WELCOME);
+		setup.setupMode();
+		border = setup.acquireBorder();
+		boardMaker = new BoardMaker(border, setup.acquireWinCondition(border));
+		players = setup.acquirePlayers();
+	}
+
 	private void playTheGame() {
```
Comment (KrzysiekJodlowski) [inline src/main/java/com/github/hongshui/ox/game/Game.java]: Cool!

File: src/main/resources/Message_pl.properties [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+WIN = \nGratulacje, to byl zwycieski ruch!\nPrzyznane punkty:\n
+DRAW = \nRemis!\n
+GAME_OVER = \nKoniec gry! Podsumowanie:\n
+
+INVALID_LANGUAGE = Podano niepoprawne dane. Prosze wybierz: en / english / pl / polski. 
+INVALID_MODE = Podano niepoprawne dane. Wybierz 1, 2 or 3. 
+INVALID_NUMBER = Podano niepoprawne dane. Prosze wybierz liczbe calkowita z zakresu od %d do %d. 
+INVALID_SYMBOL = Ten symbol zostal ju� wybrany przez innego gracza. Prosze wskaz inny. 
+INVALID_COORDINATE = Podane pole jest zajete, prosze wskaz inne.\n
```
Comment (KrzysiekJodlowski) [inline src/main/resources/Message_pl.properties]: Polish language - no polish signs?

File: src/test/java/com/github/hongshui/ox/commons/CoordinatesTest.java [hunk 9]
```diff
@@ -37,4 +37,22 @@ public void testHashCode(int row, int column) {
...
+    }
+
+    @Test(dataProvider = "sampleLengths")
+    public void testGetColumn(int row, int column) {
+        // given
+        Coordinates coordinates = new Coordinates(row, column);
+        // when
+        // then
+        assertEquals(coordinates.getColumn(), column);
+    }
```
Comment (KrzysiekJodlowski) [inline src/test/java/com/github/hongshui/ox/commons/CoordinatesTest.java]: Are you testing the setters and getters here?

File: src/test/java/com/github/hongshui/ox/datasource/InteractiveDataSourceTest.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	public void testAcquireCoordinates() {
+		// given
+		Validator validator = mock(Validator.class);
+		UserInterface ui = mock(UserInterface.class);
+		InteractiveDataSource interactiveDataSource = new InteractiveDataSource(ui, validator, SAMPLE_LENGTH);
+		int index = 0;
+		// when
+		when(validator.isValidRowOrColumn(anyString())).thenReturn(true);
+		when(validator.areValidCoordinates(any())).thenReturn(true);
+		when(ui.read()).thenReturn(String.valueOf(index + 1));
```
Comment (KrzysiekJodlowski) [inline src/test/java/com/github/hongshui/ox/datasource/InteractiveDataSourceTest.java]: Not sure if these shouldn't be in the "given" section

File: src/test/java/com/github/hongshui/ox/ui/BoardPainterTest.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+						+ " 3                                     \n"
+						+ " 4                                     \n"
+						+ " 5                                     \n"
+						+ " 6                                     \n"
+						+ " 7                                     \n"
+						+ " 8                                     \n"
+						+ " 9                                     \n"
+						+ "10                                     \n"
+						+ "11                                     \n"
+						+ "12                                     \n";
```
Comment (KrzysiekJodlowski) [inline src/test/java/com/github/hongshui/ox/ui/BoardPainterTest.java]: Could be aligned for better reading, same as below.

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java (hunk 6 above)
Comment (KrzysiekJodlowski) [reply src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: Ok I know whats that, nevermind;)

File: src/main/java/com/github/hongshui/ox/board/Lines.java (hunk 3 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/board/Lines.java]: For testing.

File: src/main/java/com/github/hongshui/ox/board/ObservableBoard.java (hunk 4 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/board/ObservableBoard.java]: Impose? What do you mean.

If you ask who implements this interface, it is Match currently. Probably should be more classes but this is a leftover from previous design which would be to difficult to fix right now :/

File: src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java (hunk 5 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/datasource/RandomDataSource.java]: Players is the class that manages list of player and it mostly just prints info, so it is a bit different that you situation

File: src/main/resources/Message_pl.properties (hunk 8 above)
Comment (hongshui) [reply src/main/resources/Message_pl.properties]: No, my computer refused to print them, so I decided that I prefer "juz" to "ju?" ;)

File: src/test/java/com/github/hongshui/ox/commons/CoordinatesTest.java (hunk 9 above)
Comment (hongshui) [reply src/test/java/com/github/hongshui/ox/commons/CoordinatesTest.java]: Yes xD I had to low test coverate...

File: src/test/java/com/github/hongshui/ox/ui/BoardPainterTest.java (hunk 11 above)
Comment (hongshui) [reply src/test/java/com/github/hongshui/ox/ui/BoardPainterTest.java]: Sadly it moves on it's own every time I push it to github. I don't know why actually...

File: estimates.adoc (hunk 1 above)
Comment (hongshui) [reply estimates.adoc]: Done

File: src/main/java/com/github/hongshui/ox/board/BoardObserver.java (hunk 2 above)
Comment (hongshui) [reply src/main/java/com/github/hongshui/ox/board/BoardObserver.java]: If I will have some time left I will try to fix those javadocs.

File: src/main/java/com/github/hongshui/ox/board/ObservableBoard.java (hunk 4 above)
Comment (KrzysiekJodlowski) [reply src/main/java/com/github/hongshui/ox/board/ObservableBoard.java]: I meant that I saw only default methods, if it's true it isn't really an interface but more a class (maybe abstract one?)

File: src/main/resources/Message_pl.properties (hunk 8 above)
Comment (KrzysiekJodlowski) [reply src/main/resources/Message_pl.properties]: https://help.ubuntu.com/stable/ubuntu-help/session-language.html.en

**Pull request merged**